ci/travis/run_tests.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -e
+
+WARNINGS="-Werror -Wall -Wextra -Wformat -Werror=format-security"
+WARNINGS_DISABLED="-Wno-unused-parameter -Wno-implicit-fallthrough -Wno-unknown-warning-option -Wno-cast-function-type"
+
+CFLAGS_STD="-g -O2 -mtune=generic -fno-omit-frame-pointer ${WARNINGS} ${WARNINGS_DISABLED}"
+LDFLAGS_STD="-Wl,-Bsymbolic-functions -Wl,-z,relro"
+
+CFLAGS_COV="-g -O0 --coverage"
+LDFLAGS_COV="--coverage"
+
+/usr/local/pgsql/bin/pg_ctl -l /tmp/logfile start
+./autogen.sh
+
+CFLAGS="${CFLAGS_STD}" LDFLAGS="${LDFLAGS_COV}" ./configure
+make -j2
+make check
+
+CFLAGS="${CFLAGS_COV}" LDFLAGS="${LDFLAGS_COV}" ./configure
+make -j2
```

Comment:
these two can be just `make -j check` - it will build everything it needs to and interleave that with check.